core/cog-host-map-handler.c

```diff
@@ -0,0 +1,299 @@
+ * @host: URI host to match
+ *
+ * Check if there is already mounted a route for the @host.
+ *
+ * Returns: Whether the route already exists.
+ */
+gboolean
+cog_host_map_handler_contains(CogHostMapHandler *self, const char *host)
+{
+    g_return_val_if_fail(COG_IS_HOST_MAP_HANDLER(self), FALSE);
+    g_return_val_if_fail(host != NULL, FALSE);
+
+    if (g_hash_table_contains(self->map, host))
+        return TRUE;
+
+    return FALSE;
+}
+
+/**
+ * cog_host_map_handler_mount:
+ * @self: a #CogHostMapHandler
+ * @host: URI host to match
+ * @handler: (transfer none): a request handler for the matched host
+ *
+ * Adds a route to the handler.
+ *
+ * Configures a route which matches @host in URI, and dispatches
+ * requests to a given @handler.
+ *
+ * Returns: Whether the route was successfully added.
+ */
+gboolean
+cog_host_map_handler_mount(CogHostMapHandler *self, const char *host, CogRequestHandler *handler)
+{
+    g_return_val_if_fail(COG_IS_HOST_MAP_HANDLER(self), FALSE);
+    g_return_val_if_fail(host != NULL, FALSE);
+    g_return_val_if_fail(COG_IS_REQUEST_HANDLER(handler), FALSE);
+
+    if (g_hash_table_contains(self->map, host))
+        return FALSE;
+
+    g_hash_table_insert(self->map, g_strdup(host), g_object_ref(handler));
+    return TRUE;
+}
+
+/**
+ * cog_host_map_handler_unmount:
+ * @self: a #CogHostMapHandler
+ * @host: URIhost for a configured route
+ *
+ * Removes a previously configured route.
+ *
+ * Removes a route that matches @host previously configured
+ * using cog_host_map_handler_mount() or
+ * cog_host_map_handler_mount_path().
+ *
+ * Returns: Whether the route was found and removed.
+ */
+gboolean
+cog_host_map_handler_unmount(CogHostMapHandler *self, const char *host)
+{
+    g_return_val_if_fail(COG_IS_HOST_MAP_HANDLER(self), FALSE);
+    g_return_val_if_fail(host != NULL, FALSE);
+
+    return g_hash_table_remove(self->map, host);
+}
+
+/**
+ * cog_host_map_handler_mount_path:
+ * @self: a #CogHostMapHandler
+ * @host: URI path host to match
+ * @base_path: Path to a local directory
+ *
+ * Adds a route to the handler pointing to a directory.
+ *
+ * This is a convenience method which configures a route
+ * matching @host in URI, and creates a new
+ * [class@Cog.DirectoryFilesHandler] for @base_path to handle requests
+ * for the route.
+ *
+ * Returns: Whether the route was successfully added.
+ */
+gboolean
+cog_host_map_handler_mount_path(CogHostMapHandler *self, const char *host, const char *base_path)
+{
+    g_return_val_if_fail(COG_IS_HOST_MAP_HANDLER(self), FALSE);
+    g_return_val_if_fail(host != NULL, FALSE);
+    g_return_val_if_fail(base_path != NULL, FALSE);
+
+    g_autoptr(GFile) path_file = g_file_new_for_path(base_path);
+    g_return_val_if_fail(cog_directory_files_handler_is_suitable_path(path_file, NULL), FALSE);
+
+    g_autoptr(CogRequestHandler) handler =
+        g_object_new(COG_TYPE_DIRECTORY_FILES_HANDLER, "base-path", path_file, "strip-components", 0);
```

Comment:
You could use `cog_directly_files_handler_new()` here for the sake of clarity. The `strip-components` property is zero by default so there is no need to use `g_object_new()` to set it during construction:

```suggestion
        cog_directory_files_handler_new(path_file);
```